File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/BusinessRepository.java
package com.ll.hotel.domain.member.member.repository;

import com.ll.hotel.domain.member.member.entity.Business;
import com.ll.hotel.domain.member.member.entity.Member;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface BusinessRepository extends JpaRepository<Business, Long> {
    Optional<Business> findByMember(Member member);
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/BusinessRepository.kt
package com.ll.hotel.domain.member.member.repository

import com.ll.hotel.domain.member.member.entity.Business
import com.ll.hotel.domain.member.member.entity.Member
import com.ll.hotel.domain.member.member.type.BusinessApprovalStatus
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BusinessRepository: JpaRepository<Business, Long> {
    fun findByMember(member: Member): Business?

    fun findByApprovalStatus(approvalStatus: BusinessApprovalStatus, pageable: Pageable): Page<Business>
}